src/components/selection_input/selection_input.xml

```diff
@@ -7,23 +7,30 @@
         t-key="range.id"
         class="o-selection-input"
         t-att-class="props.class">
-        <input
-          type="text"
-          spellcheck="false"
-          t-on-input="(ev) => this.onInputChanged(range.id, ev)"
-          t-on-focus="() => this.focus(range.id)"
-          t-on-keydown="onKeydown"
-          t-att-value="range.xc"
-          t-att-style="getColor(range)"
-          t-att-class="{
-            'o-focused' : range.isFocused,
-            'o-required': props.required,
-            'o-optional': !props.required,
-            'o-invalid': isInvalid || !range.isValidRange,
-            'text-decoration-underline': range.isFocused and state.mode === 'select-range'
-          }"
-          t-ref="{{range.isFocused ? 'focused' : 'unfocused'}}Input"
-        />
+        <div class="input-container">
+          <input
+            type="text"
+            spellcheck="false"
+            t-on-input="(ev) => this.onInputChanged(range.id, ev)"
+            t-on-focus="() => this.focus(range.id)"
+            t-on-keydown="onKeydown"
+            t-att-value="range.xc"
+            t-att-style="getColor(range)"
+            t-att-class="{
+              'o-focused' : range.isFocused,
+              'o-required': props.required,
+              'o-optional': !props.required,
+              'o-invalid': isInvalid || !range.isValidRange,
+              'text-decoration-underline': range.isFocused and state.mode === 'select-range'
+            }"
+            t-ref="{{range.isFocused ? 'focused' : 'unfocused'}}Input"
+          />
+          <span t-if="isInvalid || !range.isValidRange" class="icon" title="This range is invalid">
+            <t t-call="o-spreadsheet-Icon.ERROR">
+              <t t-set="color" t-value="'fill : #FF0000'"/>
```

Comment:
based on  https://github.com/odoo/o-spreadsheet/pull/2067/files, you can define a css rule to set the color on the parent span.